### ISSUE TITLE
NameError at `partial_view_name`  undefined local variable or method `lookbook' for #<ActionView::Base>

### ISSUE THREAD

**Issue by hjhart** [allmarkedup/lookbook#95]
Hi there! Looks like a fun project, we're just switching away from ViewComponent::Storybook and heading towards the default previews. I tried to install lookbook, and got the index page working successfully. However, when I click on one of the components in the left column, i receive this error:

```
NameError at /prism/alert_preview/default

undefined local variable or method `lookbook' for #<ActionView::Base:0x0000000005d318>
```

Where should this helper be coming from? I've installed the gem and restarted my server. 

![Screen Shot 2022-06-06 at 3 36 01 PM](https://user-images.githubusercontent.com/547981/172260687-1178dc02-d484-4f0b-b082-3c33c766e55b.png)

**Comment by allmarkedup** [comment]
Hey @hjhart - that is an odd one! Have you tried clearing your Rails template cache? Just a thought... but may help.

That helper is a actually a namespace for the path helpers - it's provided as part of the Rails engine functionality so it should be there which is why it's a bit strange.

What version of Rails are you running?

**Comment by hjhart** [comment]
Thanks for reaching out @allmarkedup!

We are running rails 6.1.6. I'm using a custom view component preview controller, perhaps that's messing up the engine helpers?

I've got some view component configuration that looks like:

```
    config.view_component.default_preview_layout = "members_plain"
    config.view_component.preview_paths << "#{Rails.root}/spec/components/previews"
```

inside our `application.rb`... other than that things look pretty vanilla.

Our `routes.rb` file has this:
```
    resources :view_component_previews, only: [:previews]
```

Which is serving that previews show action, perhaps that's messing things up? Basically, it's not going to the view_component preview controller within the gem.

**Comment by hjhart** [comment]
Update: Restarted the dev server, restarted caches with `Rails.cache.clear` in a rails console, and `rake tmp:cache:clear`, but nothing works yet. 

I even removed our custom controller and fell back to ViewComponentController from view_component. Still getting the same error.

I'll try to pick apart the engine code to better understand what's going on. Any further ideas appreciated!

**Comment by hjhart** [comment]
Things I've tried:
* Downgrading to 0.8.2. Downgrading to 0.7.4. Upgrading to 1.0.0.beta.0
![Screen Shot 2022-06-07 at 4 31 43 PM](https://user-images.githubusercontent.com/547981/172500358-1d643432-b431-4c0d-b2c2-d0f70ccb455c.png)
* Funnily enough when I remove that namespace, lookbook and the code switches from `lookbook.preview_path` to `preview_path` everything works fine.

Can you explain to me where that namespace is supposed to come from? I don't work with engines enough to understand that part.

![Screen Shot 2022-06-07 at 4 35 18 PM](https://user-images.githubusercontent.com/547981/172500799-cfd185b6-cb9c-4d0c-9803-68cc4e5d963b.png)

**Comment by hjhart** [comment]
Okay, I've figured out the difference between my application and other applications. We have a nested routes file, so the named routes you're expecting have a prefix of `members_`. Since my routes file looks like this:

```
Rails.application.routes.draw do
  draw(:members)
  draw(:staffers)
end
```

And I'm putting the `mount Lookbook::Engine, at: "/lookbook"` within the routes/members.rb file, I've now moved your expected engines namespace!

I am able to fix this by either putting the `mount Lookbook::Engine, at: "/lookbook"` in the root `routes.rb` file. 

I can alos fix this by changing the code in `_previews.html.erb` to use the namespace:

```
        src="<%= members_lookbook.preview_path(request.query_parameters.merge(lookbook_timestamp: Time.now)) %>"
```

I wonder if we should make that configurable? Or perhaps detect it from Rails perhaps? Or say "this isn't supported" explicitly in the documentation? 

Happy I figured out what's going on, not sure which path to move forward. @allmarkedup curious to hear your thoughts!

**Comment by allmarkedup** [comment]
@hjhart sorry for the slow replies, been a busy few days - but really glad you've managed to figure out what is going on.

I hadn't really considered the implications of mounting the engine within another namespace tbh. I think I should probably be explicit about only mounting the engine in the root namespace in the docs, but I'm also thinking that I should perhaps manually prefix the lookbook routes and then skip using the namespaced helper in the views (i.e. use `lookbook_preview_path` instead of `lookbook.preview_path`), to catch any other edge cases.

Thanks for the update and sorry it took up so much of your time, I'll amend the docs/views to try and mitigate the chance of something similar happening to others in the future as soon as I can.

**Comment by hjhart** [comment]
No problem. I'm not sure of a solution that fixes everything, but for now we've mounted it in the root of our routes and it's not causing any issues.

**Comment by allmarkedup** [comment]
@hjhart I know you found a solution to this but just to let you know I've just released v0.9.0 in which I've changed to using manually-prefixed route names so this should (hopefully!) not be a problem anymore. Thanks again for bringing it to my attention and for digging into it 👍
